modules/related-posts/jetpack-related-posts.php

```diff
@@ -256,60 +256,55 @@ public function get_target_html_unsupported() {
 	public function render_block_item( $related_post, $block_attributes ) {
 		$instance_id = 'related-posts-item-' . uniqid();
 		$label_id    = $instance_id . '-label';
-		?>
-		<div
-			aria-labelledby="<?php echo esc_attr( $label_id ); ?>"
-			role="menuitem"
-			data-post-id="<?php echo esc_attr( $related_post['id'] ); ?>"
-			data-post-format="<?php echo esc_attr( ! empty( $related_post['format'] ) ? $related_post['format'] : 'false' ); ?>"
-			class="jp-related-posts-i2__post"
-			id="<?php echo esc_attr( $instance_id ); ?>"
-		>
-			<a
-				href="<?php echo esc_url( $related_post['url'] ); ?>"
-				title="<?php echo esc_attr( $related_post['title'] ); ?>"
-				rel="<?php echo esc_attr( $related_post['rel'] ); ?>"
-				data-origin="<?php echo esc_attr( $related_post['url_meta']['origin'] ); ?>"
-				data-position="<?php echo esc_attr( $related_post['url_meta']['position'] ); ?>"
-				class="jp-related-posts-i2__post-link"
-				id="<?php echo esc_attr( $label_id ); ?>"
-			>
-				<?php echo esc_html( $related_post['title'] ); ?>
-			</a>
-			<?php if ( ! empty( $block_attributes['show_thumbnails'] ) && ! empty( $related_post['img']['src'] ) ) : ?>
-			<a
-				href="<?php echo esc_url( $related_post['url'] ); ?>"
-				title="<?php echo esc_attr( $related_post['title'] ); ?>"
-				rel="<?php echo esc_attr( $related_post['rel'] ); ?>"
-				data-origin="<?php echo esc_attr( $related_post['url_meta']['origin'] ); ?>"
-				data-position="<?php echo esc_attr( $related_post['url_meta']['position'] ); ?>"
-				class="jp-related-posts-i2__post-img-link"
-			>
-				<img
-					class="jp-related-posts-i2__post-img"
-					src="<?php echo esc_url( $related_post['img']['src'] ); ?>"
-					width="<?php echo esc_attr( $related_post['img']['width'] ); ?>"
-					alt="<?php echo esc_attr( $related_post['title'] ); ?>"
-				/>
-			</a>
-			<?php endif; ?>
-			<?php if ( $block_attributes['show_date'] ) : ?>
-				<div class="jp-related-posts-i2__post-date has-small-font-size">
-					<?php echo esc_html( $related_post['date'] ); ?>
-				</div>
-			<?php endif; ?>
-			<?php
-			if (
-				( $block_attributes['show_context'] ) &&
-				! empty( $related_post['context'] )
-			) :
-				?>
-				<div class="jp-related-posts-i2__post-context has-small-font-size">
-					<?php echo esc_html( $related_post['context'] ); ?>
-				</div>
-			<?php endif; ?>
-		</div>
-		<?php
+
+		$item_markup = sprintf(
+			'<ul id="%1$s" aria-labelledby="%2$s" class="jp-related-posts-i2__post" role="menuitem">',
```

Comment:
I find it a bit odd semantically to have an unordered list here, since the items in that list all relate to the same thing (one related post). Is there a reason you went with that organization?

Reply:
Yes. Those are lists of attributes, within a `nav` element. I didn't want to complicate things further by using nested lists, so this was the best compromise I found along with the appropriate role `attribute`.